spec/validators/redirect_uri_validator_spec.rb

```diff
@@ -62,13 +61,18 @@
       expect(subject).to be_valid
     end
 
-    it "accepts native redirect uri" do
+    it "accepts nonstandard oob redirect uri" do
       subject.redirect_uri = "urn:ietf:wg:oauth:2.0:oob"
       expect(subject).to be_valid
     end
 
+    it "accepts nonstandard oob:auto redirect uri" do
+      subject.redirect_uri = "urn:ietf:wg:oauth:2.0:oob:auto"
+      expect(subject).to be_valid
+    end
+
     it "accepts app redirect uri" do
-      subject.redirect_uri = "some-awesome-app://oauth/callback"
+      subject.redirect_uri = "some-awesome-app:/oauth/callback"
```

Comment:
Does it have to be a single slash? I was [looking around](https://www.oauth.com/oauth2-servers/redirect-uris/redirect-uris-native-apps/) and it seems that two slashes are used for the URI

Reply:
I think it is the recommendation in the RFC. I will confirm.

Reply:
In [rfc8252 section 7.1](https://tools.ietf.org/html/rfc8252#section-7) it states:

>Following the requirements of Section 3.2 of [RFC3986], as there is no naming authority for private-use URI scheme redirects, only a single slash ("/") appears after the scheme component.  A complete example of a redirect URI utilizing a private-use URI scheme is:

> com.example.app:/oauth2redirect/example-provider

Reply:
Where authority in rfc3986 is defined by:

authority  = [ userinfo "@" ] host [ ":" port ]

Reply:
I've test this in `irb`

```
[5] pry(main)> uri = URI.parse("some-awesome-app:/oauth/callback")
=> #<URI::Generic some-awesome-app:/oauth/callback>
[6] pry(main)> uri.scheme
=> "some-awesome-app"
[7] pry(main)> uri.path
=> "/oauth/callback"
```

It looks OK.

But I suggests to also test more normal case (which most users using for, I mean `://` style) ensuring that compatible with it and we leave a comment here about the RFC standard